pkg/barepodpatch/barepodpatch.go

```diff
@@ -0,0 +1,115 @@
+package barepodpatch
+
+import (
+	"sigs.k8s.io/yaml"
+
+	piraeusiov1 "github.com/piraeusdatastore/piraeus-operator/v2/api/v1"
+	"github.com/piraeusdatastore/piraeus-operator/v2/pkg/utils"
+)
+
+// ConvertBarePodPatch converts a patch for a bare "satellite" pod into a patch for a "linstor-satellite" daemonset
+//
+// Bare pod patches are recognized by the detected target using "kind: Pod" and "name: satellite". It converts
+// both strategic merge patches
+func ConvertBarePodPatch(patches ...piraeusiov1.Patch) []piraeusiov1.Patch {
+	for i := range patches {
+		target := patches[i].GetTarget()
+		if target == nil {
+			continue
+		}
+
+		if target.Kind != "Pod" || target.Name != "satellite" {
+			continue
+		}
+
+		if convertedPatch := convertStrategicMergePatch([]byte(patches[i].Patch)); len(convertedPatch) > 0 {
+			patches[i].Patch = string(convertedPatch)
+		} else if convertedPatch := convertJsonPatch([]byte(patches[i].Patch)); len(convertedPatch) > 0 {
+			patches[i].Patch = string(convertedPatch)
+		} else {
+			continue
```

Comment:
This is an error condition, but I guess in this case we will try to apply the unconverted patch and get an error then, so that is good enough.